test/artifacts.jl

```diff
@@ -338,7 +338,7 @@ end
         copy_test_package(project_path, "ArtifactInstallation")
         Pkg.activate(joinpath(project_path))
         add_this_pkg()
-        Pkg.add(Pkg.Types.PackageSpec(
+        Pkg.develop(Pkg.PackageSpec(
```

Comment:
@staticfloat It seems that these tests were set up incorrectly. Due to error handling `Pkg.Types.PackageSpec` is not the same as `Pkg.PackageSpec` (one is for the API one is for internal use). API functions assume the API version is used. `path` maps to `repo.url` because previously `Pkg.develop` allowed the use of git revisions. `Pkg.add` requires the package to be version controlled.

I haven't taken the time to understand the tests but I get:
```
Override.toml: Error During Test at /home/david/pkg/Pkg.jl/test/artifacts.jl:458
  Got exception outside of a @test
  LoadError: Cannot automatically install 'arty'; no download section in '/tmp/jl_pAilNr/ArtifactOverrideLoading/Artifacts.toml'
```

The artifact file seems to confirm this error message.